agent/pool/pool.go

```diff
@@ -55,6 +55,47 @@ type Conn struct {
 	clientLock sync.Mutex
 }
 
+// TimeoutConn wraps net.Conn with a default timeout.
+// readTimeout and writeTimeout apply to the subsequent operation.
```

Comment:
Is this comment accurate? `Read` and `Write` implementations look like `ReadTimeout` and `WriteTimeout` are applied first, zero'ed out, then `DefaultTimeout` is applied on subsequent operations.

Reply:
The comment is intentional but could perhaps be worded more clearly. What I meant was that setting a read/write timeout only affects the _very next_ operation, and then the timeout reverts to the default after that.